src/components/Card.js

```diff
@@ -1,49 +1,103 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
-    this._name = cardData.name;
-    this._link = cardData.link;
+  constructor({ name, link, likes, _id, userId }, myId, cardSelector, handleCardClick, handleDeleteClick, handleCardlike) {
+    this._name = name;
+    this._link = link;
+    this._likes = likes;
+    this._id = _id;
+    this._myId = myId;
+    this._userId = userId;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleCardLike = handleCardlike;
   }
 
   _getTemplate() {
-    this._cardTemplate = document.querySelector(this._cardSelector).content.querySelector(".card").cloneNode(true);
-    return this._cardTemplate;
+    return document.querySelector(this._cardSelector).content.querySelector(".card__list-item").cloneNode(true);
   }
 
-  _toggleLikeButton() {
-    this._cardElement.querySelector('.card__like-button').classList.toggle('card__like-button_active')
+  _fillCardTemplate() {
+    const imageElement = this._element.querySelector(".card__image");
+    imageElement.alt = this._name;
+    imageElement.src = this._link;
+    this._element.id = this.imageID;
+    this._element.querySelector(".card__title").textContent = this._name;
   }
 
-  _deleteCard() {
-    this._cardElement.remove();
+  _getData() {
+    const data = {
+      name: this._name,
+      link: this._link,
+    };
+    return data;
+  }
+
+  getId() {
+    return this._id;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => {
+      return like._id === this._myId;
+    });
+  }
+
+  setLikes(likes) {
+    this._likes = likes;
+    this.renderLikes();
+  }
+
+  renderLikes() {
+    this._element.querySelector(".card__like-counter").textContent = this._likes.length;
+  }
+
+  toggleLikes() {
+    if (this.isLiked()) {
+      this._element.querySelector(".card__like-button").classList.add("card__like-button_active");
+    }
+  }
+
+  _handleLikeClick() {
+    this._element.querySelector(".card__like-counter").textContent = this._likes.length;
+    if (this.isLiked()) {
+      this._element.querySelector(".card__like-button").classList.toggle("card__like-button_active");
+    }
+  }
+
+  _handleLikeIcon() {
+    this._handleCardLike({ _id: this._id, likes: this._likes });
+    this.renderLikes();
```

Comment:
You should update the likes only after successful ai response. 

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setLikes(likes) { 
   // set instance variable
    this._likes = likes; 
    this._renderLikes(); 
} 

isLiked() {
 // return true if user liked the card, otherwise false
}

_renderLikes() {
  //set  likes counter content using this._likes.length

  if (this.isLiked()) {
    // add active class to like button
  } else {
    // remove active class from like button
  }
}

generateCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  api.deleteCardLike(card._id).then(response => card.setLikes(response.likes)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```